src/processTargets/processTargets.ts

```diff
@@ -0,0 +1,260 @@
+        };
+      }
+      return startTarget.trailingDelimiter;
+    }
+    return startTarget.leadingDelimiter;
+  })();
+  const trailingDelimiter = (() => {
+    if (excludeEnd) {
+      if (endTarget.isLine) {
+        return {
+          range: new Range(
+            contentRange.end,
+            endTarget.leadingDelimiter!.range.end
+          ),
+          highlight: new Range(
+            contentRange.end,
+            endTarget.contentRange.start.translate({ lineDelta: -1 })
+          ),
+        };
+      }
+      return endTarget.leadingDelimiter;
+    }
+    return endTarget.trailingDelimiter;
+  })();
+
+  // If both objects are of the same type create a new object of the same
+  const startConstructor = Object.getPrototypeOf(startTarget).constructor;
+  const endConstructor = Object.getPrototypeOf(endTarget).constructor;
+  const constructor =
+    startConstructor === endConstructor ? startConstructor : BaseTarget;
+
+  return new constructor({
+    editor: activeTarget.editor,
+    isReversed: !isForward,
+    delimiter: anchorTarget.delimiter,
+    contentRange,
+    removalRange,
+    leadingDelimiter,
+    trailingDelimiter,
+  });
+}
+
+export function targetsToContinuousTarget(
+  anchorTarget: Target,
+  activeTarget: Target
+): Target {
+  return processContinuousRangeTarget(anchorTarget, activeTarget, false, false);
+}
+
+function processVerticalRangeTarget(
+  anchorTarget: Target,
+  activeTarget: Target,
+  excludeAnchor: boolean,
+  excludeActive: boolean
+): Target[] {
+  const isForward = calcIsForward(anchorTarget, activeTarget);
+  const delta = isForward ? 1 : -1;
+
+  const anchorPosition = isForward
+    ? anchorTarget.contentRange.end
+    : anchorTarget.contentRange.start;
+  const anchorLine = anchorPosition.line + (excludeAnchor ? delta : 0);
+  const activePosition = isForward
+    ? activeTarget.contentRange.end
+    : activeTarget.contentRange.start;
+  const activeLine = activePosition.line - (excludeActive ? delta : 0);
+
+  const results: Target[] = [];
+  for (let i = anchorLine; true; i += delta) {
+    const contentRange = new Range(
+      i,
+      anchorTarget.contentRange.start.character,
+      i,
+      anchorTarget.contentRange.end.character
+    );
+    results.push(
+      new WeakTarget({
+        editor: anchorTarget.editor,
+        isReversed: anchorTarget.isReversed,
+        position: anchorTarget.position,
+        contentRange,
+      })
+    );
+    if (i === activeLine) {
+      return results;
+    }
+  }
+}
+
+function processPrimitiveTarget(
+  context: ProcessedTargetsContext,
+  target: PrimitiveTargetDescriptor
+): Target[] {
+  const markStage = getMarkStage(target.mark);
+  let targets = markStage.run(context);
+
+  const modifierStages = [
+    // Reverse target modifies because they are in reverse order from the api. Slice is needed to create a copy or the modifiers will be in wrong order in the test recorder.
```

Comment:
```suggestion
    // Reverse target modifiers because they are in reverse order from the api. Slice is needed to create a copy or the modifiers will be in wrong order in the test recorder.
```